feat(clipboard): added wayclip support

XDream8 opened the pull request:
wayclip -> https://git.sr.ht/~noocsharp/wayclip

gpanders commented:
To satisfy the commit linter, use `feat(clipboard):` as the prefix.

Please also update the docs at `:h clipboard-tool`.

XDream8 commented:
> To satisfy the commit linter, use `feat(clipboard):` as the prefix.
> 
> Please also update the docs at `:h clipboard-tool`.

changed pr title and updated the docs

gpanders commented:
The PR title doesn't matter. You need to update the commit message.

gpanders commented on runtime/autoload/provider/clipboard.vim:
```diff
@@ -97,6 +97,12 @@ function! provider#clipboard#Executable() abort
     let s:paste['*'] = ['wl-paste', '--no-newline', '--primary']
     return 'wl-copy'
+  elseif !empty($WAYLAND_DISPLAY) && executable('waycopy') && executable('waypaste')
+    let s:copy['+'] = ['waycopy']
+    let s:paste['+'] = ['waypaste']
```
Does this need `-t text/plain` (like `wl-copy` and `wl-paste`)?

XDream8 replied:
it does not need but can be added

gpanders replied:
Please add it for consistency.

XDream8 replied:
added it!

XDream8 committed:
feat(clipboard): added wayclip support

XDream8 committed:
fix(clipboard): wayclip added mime type flag
